File: app/models/graph.rb
class Graph < ActiveRecord::Base
	belongs_to :user
	has_and_belongs_to_many :colors

	validates_presence_of :name
	validates_presence_of :category
	validates :difficulty, numericality: { only_integer: true }
	validates :gauge, numericality: { only_integer: true }
	validates :rows, presence: true, numericality: { only_integer: true }
	validates :columns, presence: true, numericality: { only_integer: true }
	validates :number_of_colors, presence: true, numericality: { only_integer: true }
	validates :layout, presence: true

end

File: app/assets/javascripts/backbone/models/graph.js
App.Models.Graph = Backbone.Model.extend({
	urlRoot: '/graphs',
	initialize: function(){
		console.log('created new graph model');
	},
	defaults: {
		user_id: null,
		name: 'My First Graph',
		category: 'My Category',
		image_url: '',
		product_image: '',
		// delete difficulty for now
		// difficulty: null,
		// gauge: 0,
		// rows: 100,
		// columns: 100,
		// delete number of colors
		// number_of_colors: 1,
		layout: '|',
		notes: 'Graph notes',
		privacy: false,
		preview: ''
	}
});
